### ISSUE TITLE
Support github token for private repositories/avoid rate-limit

### ISSUE THREAD

**Issue by jaedle** [jpillora/overseer#38]
I would really like to use this cool tool for some projects which are hosted within a private github repository. To do so there should be a chance to add a github-token to access them.

Even though this could fix problems related to rate-limits.

The configuration could look like this:
```
overseer.Run(overseer.Config{
		Program: prog,
		Fetcher: &fetcher.Github{
			User:     "jaedle",
			Repo:     "overseer",
			Interval: time.Minute,
			Token:    "super-secret-token",
		},
	})
```

Are you welcoming pull requests for this feature?

**Comment by jpillora** [comment]
Yes, adding a new `Token` option would be a welcome PR (just take care not to edit the existing methods/fields)

**Comment by Sggoodman** [comment]
It does not appear to have been added so far...
